lib/matplotlib/backend_bases.py

```diff
@@ -3604,21 +3610,28 @@ def _tool_triggered_cbk(self, event):
 
     def _add_tool_cbk(self, event):
         """Captures 'tool_added_event' and adds the tool to the toolbar"""
-        image = self._get_image_filename(event.tool.image)
+        tool, group, position = event.tool, event.data['group'], \
+                                                event.data['position']
+        if group is None:
+            return
```

Comment:
yes you are right, I did change the `in_toolbar` attribute for this and make sense. But again, the toolbar should be informed of all tools and it can decide what to include or not. The `group=None` is just a suggestion (implemented in the default backends)